crates/circuit/src/dag_node.rs

```diff
@@ -249,21 +252,17 @@ impl DAGOpNode {
             instruction,
             sort_key,
         });
-        Ok(Py::new(py, sub)?.to_object(py))
+        Py::new(py, sub)?.into_py_any(py)
     }
 
     fn __reduce__(slf: PyRef<Self>, py: Python) -> PyResult<PyObject> {
         let state = (slf.as_ref().node.map(|node| node.index()), &slf.sort_key);
-        Ok((
-            py.get_type_bound::<Self>(),
-            (
-                slf.instruction.get_operation(py)?,
-                &slf.instruction.qubits,
-                &slf.instruction.clbits,
-            ),
-            state,
-        )
-            .into_py(py))
+        let temp = (
+            slf.instruction.get_operation(py)?,
+            &slf.instruction.qubits,
+            &slf.instruction.clbits,
+        );
```

Comment:
I wouldn't mind calling `temp` something like `new_args` - the `__reduce__` tuple is complex enough as it is.